contracts/BatchNFT.sol

```diff
@@ -13,26 +13,25 @@ contract BatchNFT is ERC721, Ownable {
     using Counters for Counters.Counter;
     Counters.Counter internal _tokenIds;
 
-    uint16 public _totalSupply;
+    uint16 public totalSupply;
 
     // Base URI shared by token and contract metadata
-    string private _sharedBaseURI =
-        "ipfs://QmeEpVThsuHRUDAQccP52WV9xLa2y8LEpTnyEsPX9fp3JD/";
+    string private _sharedBaseURI = "ipfs://PLACEHOLDER_URI/";
 
     constructor(
         string memory name_,
         string memory symbol_,
-        uint16 supply_
+        uint16 supply_,
+        string memory sharedBaseURI_
     ) ERC721(name_, symbol_) {
         setSupply(supply_);
+        setSharedBaseURI(sharedBaseURI_);
     }
 
-    receive() external payable {}
-
     /**
      * @dev Withdraw all ether in the contract
      */
-    function withdrawAll(address vault_) public onlyOwner {
+    function withdrawAll(address vault_) public payable onlyOwner {
```

Comment:
added `payable` and removed `receive() external payable {}`

Reply:
Curious why this function needs to be payable?

Reply:
just for [testing](https://github.com/thematters/matters-avatar/pull/37/files#diff-ee4c354a2a46ddb89a4678af8d6e4749fefefd658910e922fe70c2ff2f965b64R84), not in use in production.